### DISCUSSION TITLE
clang-tidy configuration

### DISCUSSION THREAD

**Discussion by ChristopherPisz** [friendlyanon/cmake-init#61] [Q&A]
clang-tidy seems to be useful to run on my code, but certain messages are going to conflict with coding standard for particular groups. How do I configure clang-tidy with the setup that cmake-init has created for me?

A link to other resources on clang-tidy might be appropriate here, but I have to tie it back to the cmakepresets.json file somehow.

For an example, I'd like to to skip complaints about "warning: invalid case style for virtual method 'getId' [readability-identifier-naming]"

**Comment by friendlyanon** [comment]
`clang-tidy` is similar to `clang-format` in that they are usually configured with the `.clang-tidy` and `.clang-format` YAML files in the project root respectively.

For `clang-tidy`, you can check the documentation for its checks here: https://clang.llvm.org/extra/clang-tidy/checks/list.html
